features/url-formattering.feature

```diff
@@ -0,0 +1,127 @@
+    - de naam van een property van een resource, v.b. burgerservicenummer of
+    - de naam van een property van een gegevensgroep van een resource v.b. kinderen.burgerservicenummer
+
+Achtergrond:
+    * de Haal Centraal APIs met hun base url en base url template placeholder
+    | Naam Haal Centraal API | base url                                                         | base url template placeholder |
+    | BRP Bevragen           | https://www.haalcentraal.nl/haalcentraal/api/brp                 | {brpserverurl}                |
+    | BAG Bevragen           | https://api.bag.acceptatie.kadaster.nl/esd/huidigebevragingen/v1 | {bagserverurl}                |
+    * de persoon met burgerservicenummer "999993847"
+    * de kinderen van persoon met burgerservicenummer "999993847"
+        | burgerservicenummer kind |
+        | 999994918                |
+        | 999995121                |
+    * de verblijfplaats van persoon met burgerservicenummer "999993847"
+        | adresseerbaarobjectidentificatie |
+        | 0518010000781379                 |
+
+Rule: Met 'uri-reference' als uri format, worden 'same API links' als relatieve urls en 'externe API links' als absolute urls geformatteerd
+
+    Scenario: uri-reference is opgegeven als Accept-Uri-Format header waarde
+        Als GET "/ingeschrevenpersonen?burgerservicenummer=999993847" wordt aangeroepen met HTTP header "Accept-Uri-format"="uri-reference"
+        Dan bevat _links in de response minimaal de volgende links
+        | naam  | href                                                                                                                        | templated |
+        | base  | https://www.haalcentraal.nl/haalcentraal/api/brp                                                                            |           |
+        | self  | /ingeschrevenpersonen?burgerservicenummer=999993847                                                                         |           |
+        | adres | https://api.bag.acceptatie.kadaster.nl/esd/huidigebevragingen/v1/adressen?adresseerbaarObjectIdentificatie=0518010000781379 |           |
+        En bevat de response de volgende HTTP header
+        | naam               | waarde        |
+        | Content-Uri-Format | uri-reference |
+
+Rule: 'uri-reference' is de default uri format
+
+    Scenario: er is geen Accept-Uri-Format header in de request
+        Als GET "/ingeschrevenpersonen?burgerservicenummer=999993847" wordt aangeroepen zonder HTTP header "Accept-Uri-Format"
+        Dan bevat _links in de response minimaal de volgende links
+        | naam  | href                                                                                                                        | templated |
+        | base  | https://www.haalcentraal.nl/haalcentraal/api/brp                                                                            |           |
```

Comment:
betekent dit dat je aan _links een link "base" toevoegt? Waarvoor is dat? (als je deze link volgt krijg je een 404)

Reply:
Het is bedoeld om de response 'compleet' te maken. Bij het sturen van een http request door een client wordt impliciet de Host header toegevoegd. Mij lijkt het handig om in de response op te nemen wat als Host is gebruikt. Misschien dat 'host' hier een betere naam zou zijn geweest. Maar ik vind het ook prima om het weg te laten en pas toe te voegen als er behoefte aan is

Reply:
In rfc3986 staat in [sectie 5.1.1](https://datatracker.ietf.org/doc/html/rfc3986#section-5.1.1) dat de base URI ook in de content kan worden geëmbed. Ik denk dus dat het niet zo vreemd is om dit te doen